packages/core/test/ServerTest.ts

```diff
@@ -612,6 +612,94 @@ class WoTServerTest {
         expect(thing).to.have.property("@context").to.deep.include({"@language": "xx"});
     }
 
+    @test async "should reject if property read handler fails"() {
+        let thing = await WoTServerTest.WoT.produce({
+            title: "Thing",
+            properties: {
+                uriProp: {
+                    type: 'number'
+                }
+            }
+        });
+
+        thing.setPropertyReadHandler('uriProp', (options) => {
+            return new Promise((resolve, reject) => {
+                // Note: test reject
+                return reject('Fail expected');
+            });
+        });
+
+        let readingPossible = false;
+        try {
+            await thing.readProperty("uriProp");
+            readingPossible = true;
+        } catch (e) {
+            // as expected
+        }
+
+        if (readingPossible) {
```

Comment:
What about not using this boolean?
```js
try {
            await thing.readProperty("uriProp");
            fail("reading property 'uriProp' should throw error")
        } catch (e) {
            // as expected
            assert.ok(true);
        }
```

Reply:
This looks a bit strange I agree but throwing fail within try {} does not work.

Anyhow.. its a test only ;-)

Reply:
>This looks a bit strange I agree but throwing fail within try {} does not work
that's odd, maybe because fail actually throw an error? 🤔 .